### ISSUE TITLE
New created text file appears twice, after refresh everything is fine

### ISSUE THREAD

**Issue by dkellenb** [nextcloud/server#7499]
### Steps to reproduce
1. Create new text file using [+] icon with name `foo-bar.txt`
2. Popup opens
3. Now start typing (focus is not in the textarea, IMO a bug)
4. Type few texts
5. Wait until save text appears (`Gespeichert`)
6. Close the window

### Expected behaviour
Only one file visible (foo-bar.txt)

### Actual behaviour
Two files visible:
- {nameOfFolder}/foo-bar.txt
- {nameOfFolder}/foo-bar.txt

![selection_002](https://user-images.githubusercontent.com/9881798/33961828-bcf4dbea-e04f-11e7-95ee-7113ab1f2ec7.png)

### Server configuration
**Operating system**: CentOS 6.8 (Final)‬, centos-release-6-8.el6.centos.12.3.x86_64
**Web server:** Apache 
**Database:** MySQL 5.0.12
**PHP version:** 7.0.26
**Nextcloud version:** 12.0.4
**Updated from an older Nextcloud/ownCloud or fresh install:** older install
**Where did you install Nextcloud from:** nextcloud.com zip archive
**Signing status:**
```
No errors have been found.
```

**List of activated apps:**
```
Enabled:
  - activity: 2.5.2
  - admin_notifications: 1.0.1
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.6
  - comments: 1.2.0
  - contacts: 2.0.1
  - dav: 1.3.0
  - deck: 0.2.8
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_accesscontrol: 1.2.5
  - files_downloadactivity: 1.1.1
  - files_external: 1.3.0
  - files_markdown: 2.0.1
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - groupfolders: 1.1.0
  - impersonate: 1.0.1
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - quota_warning: 1.1.1
  - ransomware_protection: 1.0.5
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - admin_audit
  - encryption
  - tasks
  - user_external
  - user_ldap
```

**Nextcloud configuration:**
```
{
    "system": {
        "instanceid": "oc9i4xlxxw2s",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.***DOMAIN***.ch"
        ],
        "datadirectory": "\/home\/httpd\/vhosts\/cloud.DOMAIN.ch\/data",
        "overwrite.cli.url": "https:\/\/cloud.***DOMAIN***.ch",
        "dbtype": "mysql",
        "version": "12.0.4.3",
        "dbname": "nextcloud_prod",
        "dbhost": "localhost:3306",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 0,
        "maintenance": false,
        "theme": ""
    }
}
```

**Are you using encryption:** no

### Client configuration
**Browser:** Chromium
**Operating system:** Debian Stretch

#### Nextcloud log (data/nextcloud.log)
```
Error: Call to a member function getPath() on null
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/Preview/GeneratorHelper.php - line 53: OC\Preview\GeneratorHelper->getViewAndPath(NULL)
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/Preview/Generator.php - line 162: OC\Preview\GeneratorHelper->getThumbnail(Object(OC\Preview\TXT), Object(OC\Files\Node\File), 2048, 2048)
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/Preview/Generator.php - line 110: OC\Preview\Generator->getMaxPreview(Object(OC\Files\SimpleFS\SimpleFolder), Object(OC\Files\Node\File), 'text/plain')
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/PreviewManager.php - line 201: OC\Preview\Generator->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill', 'text/plain')
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/core/Controller/PreviewController.php - line 113: OC\PreviewManager->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill')
[internal function] OC\Core\Controller\PreviewController->getPreview(Object(OC\Files\Node\File), 32, 32, false, false, 'fill')
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/AppFramework/Http/Dispatcher.php - line 160: call_user_func_array(Array, Array)
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/AppFramework/Http/Dispatcher.php - line 90: OC\AppFramework\Http\Dispatcher->executeController(Object(OC\Core\Controller\PreviewController), 'getPreview')
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/AppFramework/App.php - line 114: OC\AppFramework\Http\Dispatcher->dispatch(Object(OC\Core\Controller\PreviewController), 'getPreview')
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('OC\\Core\\Control...', 'getPreview', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/private/Route/Router.php - line 299: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/lib/base.php - line 1004: OC\Route\Router->match('/core/preview.p...')
/home/httpd/vhosts/cloud.DOMAIN.ch/httpdocs/index.php - line 48: OC handleRequest()
{main}
```

**Comment by MorrisJobke** [comment]
cc @icewind1991 @danxuliu

**Comment by MorrisJobke** [comment]
For me this happens, when I create a file and immediately favourite it after creation (without a page refresh in between).

**Comment by MorrisJobke** [comment]
cc @rullzer for the preview error.

**Comment by heinrichmartin** [comment]
Is this a duplicate of #4869?

**Comment by dnutan** [comment]
Same behavior as described by @dkellenb, but on Nextcloud 13.0.0.14
The row duplication is sometimes triggered also after clicking the share icon on a newly created text file.

**Comment by rullzer** [comment]
It is fixed in 13.0.1
